lightkurve/correctors/designmatrix.py

```diff
@@ -224,56 +252,101 @@ def append_constant(self, prior_mu=0, prior_sigma=np.inf):
+                                 ''.format(self.shape[1]))
+        if self.prior_sigma is not None:
+            if len(self.prior_sigma) != self.shape[1]:
+                raise ValueError('`prior_sigma` must have shape {}'
+                                 ''.format(self.shape[1]))
+            if np.any(np.asarray(self.prior_sigma) <= 0):
+                raise ValueError('`prior_sigma` values cannot be smaller than '
+                                 'or equal to zero')
+
+    def validate(self, rank=True):
+        """Raises a `LightkurveWarning` if the matrix has a low rank, or priors that are the wrong shape."""
+        self._validate()
 
     @property
     def rank(self):
         """Matrix rank computed using `numpy.linalg.matrix_rank`."""
         return np.linalg.matrix_rank(self.values)
 
-    @property
-    def columns(self):
-        """List of column names."""
-        return list(self.df.columns)
-
     @property
     def shape(self):
         """Tuple specifying the shape of the matrix as (n_rows, n_columns)."""
-        return self.df.shape
+        return self.X.shape
 
     @property
     def values(self):
         """2D numpy array containing the matrix values."""
         return self.df.values
 
     def __getitem__(self, key):
-        return self.df[key]
+        return self.df[key].values
 
     def __repr__(self):
         return '{} DesignMatrix {}'.format(self.name, self.shape)
 
+    def to_sparse(self):
+        """ Convert a dense DesignMatrix object to a SparseDesignMatrix
+
+        The values of this design matrix will be converted to a
+        `scipy.sparse.csr_matrix`, which stores the values in a
+        lower memory matrix. This is not recommended for dense matrices.
+        """
+        return SparseDesignMatrix(csr_matrix(self.values), name=self.name,
+                            columns=self.columns, prior_mu=self.prior_mu,
+                             prior_sigma=self.prior_sigma)
+
+    def join(self, matrix):
+        """ Join two designmatrices, return a design matrix collection """
+        return DesignMatrixCollection([self, matrix])
+
+
 
 class DesignMatrixCollection():
     """A set of design matrices."""
     def __init__(self, matrices):
-        self.matrices = matrices
+        if np.any([issparse(m.X) for m in matrices]):
+            warnings.warn(('Some matrices are `SparseDesignMatrix` objects. '
+                            'Sparse matrices will be converted to dense matrices.'), LightkurveWarning)
+            dense_matrices = []
+            for m in matrices:
+                if isinstance(m, SparseDesignMatrix):
+                    dense_matrices.append(m.copy().to_dense())
+                else:
+                    dense_matrices.append(m)
+            self.matrices = dense_matrices
+        else:
+            self.matrices = matrices
+        self.X = np.hstack(tuple(m.X for m in self.matrices))
+        self._child_class = DesignMatrix
+        self.validate()
```

Comment:
It's fairly obvious to use now, but I'd add 2 or 3 lines of comments in this constructor along the lines of:

`# This collection is designed for dense matrices, so we raise a warning if a SparseDesignMatrix is passed`

and

`# If a sparse matrix is parsed, we convert it do a dense one for user convenience`